docs/data-types.md

```diff
@@ -0,0 +1,24 @@
+# Data types 
+Data types present a number of thorny edge cases when dealing with different language standards and text file representations. This document lays out a consistent approach to data types, coercion, and error handling that is informed by pandas, Arquero, and the JavaScript and Python language behaviors. In general, we try to respect language defaults unless there are clear issues that the data wrangling context presents. We also try to establish reasonable defaults that can be observed across implementations with minimal configuration or UX, since we are building tools for non-data-scientists.
+
+## Common tricky use cases:
+- Text-based data files may contain strings that represent primitive values. Parsing these files should respect the data file's intent even if it overrides default language behavior. The most common example of this is probably boolean data columns with the values "true" and "false". JavaScript will naturally parse any non-empty string as `true`, so "false" -> `true`. A similar situation has been observed with "null".
+- Dates can be represented in a wide variey of formats, and parsing/guessing implementations differ by platform and library.
+  - `new Date()` in JavaScript is problematic, and may also conflict with pandas' [date guessing](https://pandas.pydata.org/pandas-docs/stable/user_guide/io.html?highlight=date#date-handling).
+- Autotomatic type discovery for columns is performed by both Arquero and pandas, but may have different results.
+- Some verbs can only be performed on certain data types, and other verbs can work with different data types but have different operators available. For example:
+  - [bin](./verbs/bin.md) requires numeric input types.
+  - [filter](./verbs/filter.md) requires different comparison operators depending on type (e.g., string 'contains' versus numeric 'less than').
+- Python does not have `undefined`, but it is commonly used as a return value in JavaScript for optional properties, potentially causing cross-platform/serialization confusion.
+
+## Rules
```

Comment:
SDS has the notion of sensitive zeros, maybe we can try to use something similar here?

What I mean is, let's say we have a string column, for that, we might have strings with "null" or "None" represented as actual strings. Generally, we might want to translate that to empty values, although, there might be some edge cases where we might not. Can this be made an option when loading the data?

Reply:
We definitely need more robust data loading, and I've created a ticket to address this. Pandas has default null strings, but it is configurable.